### PR TITLE
Make reports compatible with Postgres and MS SQL, add unit tests

### PR DESCRIPTION
**Pull request by edlinklater**
The fix for reports implemented in #164 unfortunately broke Postgres and MS SQL support by using DATE_FORMAT and insufficient escaping. Additionally, unit tests for reports have been added to catch such issues in future.

### CODE REVIEW

**Comment by assertchris** [comment]
This looks great, thanks Ed!

**Comment by dhensby** [comment]
fantastic

File: code/reports/ForumReport.php [hunk 1]
```diff
@@ -3,46 +3,49 @@
...
 	public function sourceRecords($params = array()) {
 		$membersQuery = new SQLQuery();
-		$membersQuery->setFrom('Member');
-		$membersQuery->selectField('DATE_FORMAT("Created", \'%Y %M\')', 'Month');
-		$membersQuery->selectField('COUNT("Created")', 'Signups');
-		$membersQuery->setGroupBy('Month');
-		$membersQuery->setOrderBy('Created', 'DESC');
+		$membersQuery->setFrom('"Member"');
+		$membersQuery->setSelect(array(
+			'Month' => DB::getConn()->formattedDatetimeClause('"Created"', '%Y-%m'),
```
Comment (tractorcow) [inline code/reports/ForumReport.php]: One day we relly should create a datetime specific SearchFilter.
```php
array(
	`Created:Year` => $year,
	`Created:Month:LessThan` => $month
)
```

Or something. :)

Comment (dhensby) [inline code/reports/ForumReport.php]: We definitely should!

**Pull request merged**